### PR TITLE
v6.3.0

### PR DESCRIPTION
**Pull request by LukasTy**
### Release the packages

- [ ] Checkout the last version of the working branch
- [ ] Make sure you have the latest dependencies installed: `yarn`.
- [ ] Build the packages: `yarn release:build`.
- [ ] Release the versions on npm: `yarn release:publish` (you need your 2FA device).
- [ ] Push the newly created tag: `yarn release:tag`.

### Publish the documentation

The documentation must be updated on the `docs-vX` branch (`docs-v4` for `v4.X` releases, `docs-v5` for `v5.X` releases, ...)

- [ ] Push the working branch on the documentation release branch to deploy the documentation with the latest changes.

```sh
git push upstream master:docs-v6 -f
```

You can follow the deployment process [on the Netlify Dashboard](https://app.netlify.com/sites/material-ui-x/deploys?filter=docs-v6)
Once deployed, it will be accessible at https://material-ui-x.netlify.app/ for the `docs-v6` deployment.

### Publish GitHub release

- [ ] After documentation is deployed, publish a new release on [GitHub releases page](https://github.com/mui/mui-x/releases)

### CODE REVIEW

File: packages/eslint-plugin-material-ui/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "eslint-plugin-material-ui",
-  "version": "6.0.5",
+  "version": "6.3.0",
```
Comment (LukasTy) [inline packages/eslint-plugin-material-ui/package.json:3]: The eslint dependency of this package has been updated. Proceeded with bumping the version just for the same of consistency. Do you agree?

**Comment by LukasTy** [comment]
### Clarification

There are certain packages that have been detected by lerna as candidates for release because of changes (package bumps (x-license-pro) or unrelated tsconfig change (x-codemod)).
Do we always release packages in such cases or employ logic: if it does not affect the end-user—refrain from making a pointless release? 🤔

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,68 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 6.3.0
+
+_Apr 27, 2023_
```
Comment (alexfauquette) [inline CHANGELOG.md:8]: Except if you want to do the release tonight
```suggestion
_Apr 28, 2023_
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,68 @@
...
+- [docs] Fix error in `minDateTime` `validation` page section (#8777) @LukasTy
+- [docs] Update custom field pickers using theme scoping (#8609) @siriwatknp
+- [docs] Use community version of data grid for column grouping demo (#7346) @ASchwad
+- [docs] Use new `slots` / `slotProps` props in the pickers migration guide (#8341) @flaviendelangle
+
+### Core
+
+- [core] Cleanup picker tests (#8652) @flaviendelangle
+- [core] Use `adapter.lib` instead of `adapterName` in `describeAdapters` (#8779) @flaviendelangle
+- [charts] Adapt line and scatter the band scale (#8701) @alexfauquette
```
Comment (alexfauquette) [inline CHANGELOG.md:65]:
```suggestion
- [charts] Adapt line and scatter plot to the "band" scale type (#8701) @alexfauquette
```

**Review by alexfauquette — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,68 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 6.3.0
+
+_Apr 27, 2023_
+
+We'd like to offer a big thanks to the 13 contributors who made this release possible. Here are some highlights ✨:
+
+- 🚀 Add new `DigitalClock` desktop [time picking experience](https://mui.com/x/react-date-pickers/digital-clock/) (#7958) @LukasTy
```
Comment (flaviendelangle) [inline CHANGELOG.md:12]: Could be worth adding a screenshot here

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,68 @@
...
+- [l10n] Improve Chinese (zh-CN) locale (#8753) @SakumyZ
+- [l10n] Improve French (fr-FR) locale (#8704) @Jul13nT
+- [DataGrid] Add Joy UI slots (button and switch) (#8699) @siriwatknp
+- [DataGrid] Fix aggregation label alignment (#8694) @joserodolfofreitas
+
+### `@mui/x-date-pickers@v6.3.0` / `@mui/x-date-pickers-pro@v6.3.0`
+
+#### Changes
+
+- [fields] Always add leading zeroes on digit sections (#8527) @flaviendelangle
```
Comment (flaviendelangle) [inline CHANGELOG.md:34]: I think that one is worth highlighting, it's a change that people will notice

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,68 @@
...
+
+#### Changes
+
+- [DataGrid] Add overlay classes to `gridClasses` (#8686) @lindapaiste
+- [DataGrid] Fix 'ResizeObserver loop limit exceeded' error (#8744) @m4theushw
+- [DataGridPremium] Fix infinite loop when updating grouped rows (#8693) @cherniavskii
+- [DataGridPro] Fix error after updating `columns` and `columnGroupingModel` at once (#8730) @cherniavskii
+- [l10n] Improve Chinese (zh-CN) locale (#8753) @SakumyZ
+- [l10n] Improve French (fr-FR) locale (#8704) @Jul13nT
+- [DataGrid] Add Joy UI slots (button and switch) (#8699) @siriwatknp
```
Comment (flaviendelangle) [inline CHANGELOG.md:27]: Move the two last one up to follow alphabetical order ?

File: CHANGELOG.md (hunk 4 above)
Comment (joserodolfofreitas) [inline CHANGELOG.md:12]:
```suggestion
- 🚀 New [time-picking UI](https://mui.com/x/react-date-pickers/digital-clock/) to replace the Time Clock on desktops (#7958) @LukasTy
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,6 +3,68 @@
...
+- [fields] Pass the `readOnly` prop to `InputProps` instead of `inputProps` (#8659) @flaviendelangle
+- [pickers] Add missing export for `caES` locale (#8782) @flaviendelangle
+- [pickers] Add new `DigitalClock` desktop time picking experience (#7958) @LukasTy
+- [pickers] Do not use `instanceOf DateTime` in `AdapterLuxon` (#8734) @flaviendelangle
+- [pickers] Fix date calendar `selected` & `disabled` day style (#8773) @LukasTy
+- [pickers] Migrate `AdapterDateFns` to our repository (#8736) @flaviendelangle
+- [pickers] Migrate `AdapterLuxon` to our repository (#8600) @flaviendelangle
+- [pickers] Migrate `AdapterMomentJalaali` and `AdapterDateFnsJalali` to our repository (#8741) @flaviendelangle
+- [pickers] Migrate `AdapterMoment` to our repository (#8700) @flaviendelangle
+- [pickers] Migrate `MomentHijri` to our repository (#8776) @flaviendelangle
```
Comment (flaviendelangle) [inline CHANGELOG.md:44]:
```suggestion
- [pickers] Migrate `AdapterMomentHijri` to our repository (#8776) @flaviendelangle
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -3,6 +3,68 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 6.3.0
+
+_Apr 27, 2023_
+
+We'd like to offer a big thanks to the 13 contributors who made this release possible. Here are some highlights ✨:
```
[the remark this answers is not in the document]
Comment (LukasTy) [reply CHANGELOG.md]: Merged another l10n PR, hence updated to 15. 😉

File: CHANGELOG.md [hunk 9]
```diff
@@ -3,6 +3,68 @@
...
+- 🌍 Improve Chinese (zh-CN) and French (fr-FR) locales
+- 🐞 Bugfixes
+- 📚 Documentation improvements
+
+### `@mui/x-data-grid@v6.3.0` / `@mui/x-data-grid-pro@v6.3.0` / `@mui/x-data-grid-premium@v6.3.0`
+
+#### Changes
+
+- [DataGrid] Add overlay classes to `gridClasses` (#8686) @lindapaiste
+- [DataGrid] Fix 'ResizeObserver loop limit exceeded' error (#8744) @m4theushw
```
[the remark this answers is not in the document]
Comment (LukasTy) [reply CHANGELOG.md:27]: Added in different place—sorted alphabetically.

File: CHANGELOG.md (hunk 7 above)
Comment (LukasTy) [reply CHANGELOG.md]: Updated and moved location to follow alphabetical order.

File: CHANGELOG.md (hunk 6 above)
Comment (LukasTy) [reply CHANGELOG.md]: Done.

File: CHANGELOG.md (hunk 4 above)
Comment (LukasTy) [reply CHANGELOG.md]: What do you think about the added GIF? Will it be good enough? 🤔

File: CHANGELOG.md (hunk 5 above)
Comment (LukasTy) [reply CHANGELOG.md:40]: Added a highlight, feel free to suggest an improvement. 🤔

File: CHANGELOG.md [hunk 10]
```diff
@@ -3,6 +3,75 @@
...
+
+_Apr 28, 2023_
+
+We'd like to offer a big thanks to the 15 contributors who made this release possible. Here are some highlights ✨:
+
+- 🚀 New [time-picking UI](https://mui.com/x/react-date-pickers/digital-clock/) designed for desktops (#7958) @LukasTy
+
+  <img src="https://user-images.githubusercontent.com/4941090/235072007-de39a397-e4a4-4c98-8e10-5ee4ad440108.gif" width="494" />
+
+- ✨ Picker fields [will include a leading zero](https://mui.com/x/react-date-pickers/adapters-locale/#respect-leading-zeros-in-fields) on digit sections (#8527) @flaviendelangle
```
Comment (flaviendelangle) [inline CHANGELOG.md:16]:
```suggestion
- ✨ Picker fields [now always include a leading zero](https://mui.com/x/react-date-pickers/adapters-locale/#respect-leading-zeros-in-fields) on digit sections (#8527) @flaviendelangle
```

**Review by flaviendelangle — APPROVED** [review]

**Pull request merged**